RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,185 @@
+    init(configuration: CustomerCenterConfigData,
+         subscriptionInformation: SubscriptionInformation) {
+        self.configuration = configuration
+        self.subscriptionInformation = subscriptionInformation
+        self.purchasesProvider = ManageSubscriptionPurchases()
+        state = .success
+    }
+
+    func loadScreen() async {
+        do {
+            try await loadSubscriptionInformation()
+            await loadCustomerCenterConfig()
+            self.state = .success
+        } catch {
+            self.state = .error(error)
+        }
+    }
+
+    func loadSubscriptionInformation() async throws {
+        let customerInfo = try await purchasesProvider.customerInfo()
+        guard let currentEntitlementDict = customerInfo.entitlements.active.first,
+              let subscribedProductID = customerInfo.activeSubscriptions.first,
+              let subscribedProduct = await purchasesProvider.products([subscribedProductID]).first else {
+            Logger.warning(Strings.could_not_find_subscription_information)
+            throw CustomerCenterError.couldNotFindSubscriptionInformation
+        }
+        let currentEntitlement = currentEntitlementDict.value
+
+        // swiftlint:disable:next todo
+        // TODO: support non-consumables
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .medium
+        self.subscriptionInformation = SubscriptionInformation(
+            title: subscribedProduct.localizedTitle,
+            durationTitle: subscribedProduct.subscriptionPeriod?.durationTitle ?? "",
+            price: subscribedProduct.localizedPriceString,
+            nextRenewalString: currentEntitlement.expirationDate.map { dateFormatter.string(from: $0) } ?? nil,
+            willRenew: currentEntitlement.willRenew,
+            productIdentifier: subscribedProductID,
+            active: currentEntitlement.isActive
+        )
+    }
+
+    func loadCustomerCenterConfig() async {
+        self.configuration = CustomerCenterConfigTestData.customerCenterData
+    }
+
+    #if os(iOS) || targetEnvironment(macCatalyst)
+    func handleAction(for path: CustomerCenterConfigData.HelpPath) {
+        switch path.type {
+        case .missingPurchase:
+            self.showRestoreAlert = true
+        case .refundRequest:
+            Task {
+                guard let subscriptionInformation = self.subscriptionInformation else { return }
+                let productId = subscriptionInformation.productIdentifier
+                let status = try await purchasesProvider.beginRefundRequest(forProduct: productId)
+                switch status {
+                case .error:
+                    self.refundRequestStatus = "Error when requesting refund, try again"
+                case .success:
+                    self.refundRequestStatus = "Refund granted successfully!"
+                case .userCancelled:
+                    self.refundRequestStatus = "Refund canceled"
```

Comment:
We probably will need to localize these? I'm seeing that their usage is combined with another string so we would need to localize this (with `String(localized: "")` and the other text where it's used (in order to avoid localizing all the combinations of these

Reply:
Separately, but I don't really like how these are localized with SwiftUI, since we don't have any compilation-time check that all the localizations are there... Maybe we should extract all these to strings enums so we have it better organized...

Reply:
right, I agree with the localization... I am surprised xcode doesn't warn or anything. We should come back to this because if I am not wrong it's the same mechanism in Paywalls right?